lib/explorer/data_frame.ex

```diff
@@ -5437,6 +5430,212 @@ defmodule Explorer.DataFrame do
+  exact match is matched with the closest earlier date from `gdp`:
+
+      iex> gdp = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-01-01], ~D[2017-01-01], ~D[2018-01-01], ~D[2019-01-01],~D[2020-01-01]],
+      ...>   gdp: [4164, 4411, 4566, 4696, 4827]
+      ...> )
+      iex> population = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-03-01], ~D[2018-08-01], ~D[2019-01-01]],
+      ...>   population: [82.19, 82.66, 83.12]
+      ...> )
+      iex> Explorer.DataFrame.join_asof(population, gdp, strategy: :backward)
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        date date [2016-03-01, 2018-08-01, 2019-01-01]
+        population f64 [82.19, 82.66, 83.12]
+        gdp s64 [4164, 4566, 4696]
+      >
+
+  Note how:
+
+  * date `2016-03-01` from `population` is matched with `2016-01-01` from `gdp`;
+  * date `2018-08-01` from `population` is matched with `2018-01-01` from `gdp`.
+
+  If we instead use `strategy: :forward`, then each date from `population` which doesn’t have an
+  exact match is matched with the closest later date from `gdp`:
+
+      iex> gdp = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-01-01], ~D[2017-01-01], ~D[2018-01-01], ~D[2019-01-01],~D[2020-01-01]],
+      ...>   gdp: [4164, 4411, 4566, 4696, 4827]
+      ...> )
+      iex> population = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-03-01], ~D[2018-08-01], ~D[2019-01-01]],
+      ...>   population: [82.19, 82.66, 83.12]
+      ...> )
+      iex> Explorer.DataFrame.join_asof(population, gdp, strategy: :forward)
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        date date [2016-03-01, 2018-08-01, 2019-01-01]
+        population f64 [82.19, 82.66, 83.12]
+        gdp s64 [4411, 4696, 4696]
+      >
+
+  Note how:
+
+  * date `2016-03-01` from `population` is matched with `2017-01-01` from `gdp`;
+  * date `2018-08-01` from `population` is matched with `2019-01-01` from `gdp`.
+
+  Finally, `strategy: :nearest` gives us a mix of the two results above, as each date from
+  `population` which doesn’t have an exact match is matched with the closest date from `gdp`,
+  regardless of whether it’s earlier or later:
+
+      iex> gdp = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-01-01], ~D[2017-01-01], ~D[2018-01-01], ~D[2019-01-01],~D[2020-01-01]],
+      ...>   gdp: [4164, 4411, 4566, 4696, 4827]
+      ...> )
+      iex> population = Explorer.DataFrame.new(
+      ...>   date: [~D[2016-03-01], ~D[2018-08-01], ~D[2019-01-01]],
+      ...>   population: [82.19, 82.66, 83.12]
+      ...> )
+      iex> Explorer.DataFrame.join_asof(population, gdp, strategy: :nearest)
+      #Explorer.DataFrame<
+        Polars[3 x 3]
+        date date [2016-03-01, 2018-08-01, 2019-01-01]
+        population f64 [82.19, 82.66, 83.12]
+        gdp s64 [4164, 4696, 4696]
+      >
+
+  The `by` argument allows joining on another column first, before the asof join. In this example we join by `country` first, then asof join by date, as above.
```

Comment:
```suggestion
  The `by` argument allows left-joining on another column (or columns) first,
  before the asof join. In this example we left-join by `country` first, then
  asof join by `date`, as above.
```

I think this helps clarify the ambiguity of the word "join".